### ISSUE TITLE
Still has ads

### ISSUE THREAD

**Issue by t0stiman** [FriedChickenButt/youtube-webos#7]
I still get an ad before the actual video is started. Happens both in the emulator and on the TV.
Does it still work for you?

**Comment by FriedChickenButt** [comment]
Youtube receives ADs through multiple domains, most of which can be caught using the regex defined in [this code](https://github.com/FriedChickenButt/youtube-webos/blob/main/webOSUserScripts/userScript.js#L2). 

But, youtube also uses a few of its own domains (which are used for the actual videos) to get the ADs. Therefore, blocking these domains would break the ability of youtube app to play many videos.

I haven't tried to dive deep into youtube.com/tv logic to accurately determine how the ADs are downloaded, which I plan to take up at a later date, but the regex linked earlier blocks most of the ADs.

However, I highly encourage to raise a PR/Issue if anyone has more information on how to make the AD blocking better.

**Comment by t0stiman** [comment]
ah, okay. this app doesn't really add anything for me then, I think, because I already block ad domains with a network-wide ad-blocker (pi-hole).

**Comment by FriedChickenButt** [comment]
Is it possible to have Ublock as a tampermonkey script? If yes, integrating would be very simple. If not, I would need to wrap a lot of chrome extension APIs. I haven't had a look at how Ublock works, but I'm going to take this up soon.

**Comment by Informatic** [comment]
I think it'd be best to just find the actual ublock rules that implement youtube adblocking and reimplement them (or make a minimal viable implementation of said rules parser / interpreter). That shouldn't be very hard.

**Comment by Informatic** [comment]
Also - is ublock origin also effective against youtube.com/tv ads?

**Comment by Informatic** [comment]
effectively youtube.com/tv adblocking by ublock origin is handled by the following rule from https://raw.githubusercontent.com/uBlockOrigin/uAssets/master/filters/filters.txt

```
youtube.com,youtube-nocookie.com##+js(json-prune, [].playerResponse.adPlacements [].playerResponse.playerAds playerResponse.adPlacements playerResponse.playerAds adPlacements playerAds)
```

This in practice calls the following snippet with all the arguments:
https://github.com/gorhill/uBlock/blob/bfdc81e9e400f7b78b2abc97576c3d7bf3a11a0b/assets/resources/scriptlets.js#L365-L470

After preliminary research, following patch ran manually via developer console seems to remove ads in my firefox browser with user-agent spoofed as webOS:
```
p = JSON.parse; window.JSON.parse = function() { const r = p.apply(this, arguments); if (r.adPlacements) {console.info('patching', r); r.adPlacements=[];} return r; }
```

**Comment by FriedChickenButt** [comment]
Thank you @Informatic for doing the research, I will most probably implement it in the current weekend.

**Comment by Informatic** [comment]
This will be closed in #8 / #9

**Comment by Informatic** [comment]
This has been fixed in c6f4180516d9d43d092ca6db4355b33ef529346a merged already.

**Comment by t0stiman** [comment]
ok
